### PR TITLE
Kris

### PR DESCRIPTION
**Pull request by tmagicturtle**
I have tested this and it works. I am not confident that the way that I handled the randomized mega evolution was the most ideal way of doing it, however.

### CODE REVIEW

File: data/mods/ssb/statuses.js [hunk 1]
```diff
@@ -772,6 +772,29 @@ let BattleStatuses = {
 			this.add(`c|@KingSwordYT|BUAAAAAA IYA AYÚDAME :(`);
 		},
 	},
+	kris: {
+		noCopy: true,
+		onStart(pokemon) {
+			this.add(`c|+Kris|glhf`);
```
Comment (HoeenCoder) [inline data/mods/ssb/statuses.js:778]: Add an illusion check after the message but before the innate.

**Review by HoeenCoder — COMMENTED** [review]
Secondary not included on the custom move.

Concerning the mega stone, I'm not sure how this is working actually. I feel like the client would be unable to tell that rotom can mega evolve with the current implementation but it seems to work according to you.

Looking into modifying `canMegaEvo` appears to be an easy solution.
```javascript
// existing code
canMegaEvo(pokemon) {
		let altForme = pokemon.baseTemplate.otherFormes && this.getTemplate(pokemon.baseTemplate.otherFormes[0]);
		let item = pokemon.getItem();
		if (altForme && altForme.isMega && altForme.requiredMove && pokemon.baseMoves.includes(toID(altForme.requiredMove)) && !item.zMove) return altForme.species;
		if (item.megaEvolves !== pokemon.baseTemplate.baseSpecies || item.megaStone === pokemon.species) {
			return null;
		}
		return item.megaStone;
	},
```
Specifically you will need to make the `if (item.megaEvolves !== pokemon.baseTemplate.baseSpecies || item.megaStone === pokemon.species) {` support an array, and when `return`ing, return one of the possible mega evos from the array.

You will also need to modify Thunderstone in items.js, use magmarizer for an example. Typescript will not like you using an array in the item.megaStone property, so you will need to // @ ts-ignore that because we don't want to edit the TS files for gens 1-7 for this.

**Comment by tmagicturtle** [comment]
Could have sworn I already left a comment - sorry if this appears twice. It wouldn't work as committed because I forgot to add the changes I made to items.js. However, I simply put ``megaStone: "Rotom-Wash"`` and did not make changes to allow an array in its place. It worked as expected, though, and ignored the string.

File: data/mods/ssb/scripts.js [hunk 2]
```diff
@@ -107,15 +107,22 @@ let BattleScripts = {
 		}
 		if (noLock && pokemon.volatiles.lockedmove) delete pokemon.volatiles.lockedmove;
 	},
+	// Modded to allow arrays as Mega Stone options
+	canMegaEvo(pokemon) {
+		let altForme = pokemon.baseTemplate.otherFormes && this.getTemplate(pokemon.baseTemplate.otherFormes[0]);
+		let item = pokemon.getItem();
+		if (altForme && altForme.isMega && altForme.requiredMove && pokemon.baseMoves.includes(toID(altForme.requiredMove)) && !item.zMove) return altForme.species;
+		if (item.megaEvolves !== pokemon.baseTemplate.baseSpecies || item.megaStone === pokemon.species) {
```
Comment (HoeenCoder) [inline data/mods/ssb/scripts.js:115]: Also add a check to ensure pokemon.species isnt in the array (if it is one) so that you cant try to mega a mega

**Review by HoeenCoder — COMMENTED** [review]
Did not notice this on my first pass, Ectoplasm should fail if the user dosen't have a secondary type. While this is not an issue for rotom, if its called by a pokemon that dosen't have a secondary type, it should fail.

**Pull request merged**